holmes/plugins/toolsets/prometheus/data_compression.py

```diff
@@ -0,0 +1,348 @@
+    else:
+        while most_common_label and match_count > 1:
+            current_group_labels = set()
+            current_group: list[CompressedMetric] = []
+            current_group_labels.add(most_common_label)
+            unmatched_metrics = []
+            for metric_data in metrics_to_process:
+                if most_common_label in metric_data.labels:
+                    current_group.append(metric_data)
+                else:
+                    unmatched_metrics.append(metric_data)
+
+            all_group_labels = current_group_labels.union(globally_common_labels)
+            most_common_label, match_count = find_most_common_label(
+                metrics=current_group, ignore_label_set=all_group_labels
+            )
+
+            # Keep aggregating all labels that are common with the current group.
+            while match_count == len(current_group) and most_common_label is not None:
+                current_group_labels.add(most_common_label)
+                all_group_labels = current_group_labels.union(globally_common_labels)
+                most_common_label, match_count = find_most_common_label(
+                    metrics=current_group, ignore_label_set=all_group_labels
+                )
+
+            # We're done with our group as we found no more common labels.
+            # 1. Remove the common labels from all metrics in this group
+            # 2. Recurse to further group metrics within this group
+            for metric_in_group in current_group:
+                all_group_labels = current_group_labels.union(globally_common_labels)
+                remove_labels(metric=metric_in_group, remove_labels=all_group_labels)
+
+            groups.append(
+                Group(common_labels=current_group_labels, metrics=current_group)
+            )
+
+            most_common_label, match_count = find_most_common_label(
+                metrics=unmatched_metrics, ignore_label_set=globally_common_labels
+            )
+            metrics_to_process = unmatched_metrics
+
+    for metric in unmatched_metrics:
+        remove_labels(metric=metric, remove_labels=globally_common_labels)
+        # prepend instead of append so that unique metrics are closer to common labels than grouped metrics
+        # I 'guess' it may help the LLM in making sense of the hierarchy
+        groups.insert(0, metric)
+
+    if globally_common_labels and len(groups) > 1:
+        parent_group = Group(common_labels=globally_common_labels, metrics=groups)
+        return [parent_group]
+    else:
+        return groups
+
+
+def set_default(obj):
+    if isinstance(obj, set):
+        return list(obj)
+    raise TypeError
+
+
+def summarize_metrics(metrics_to_process: list[CompressedMetric]) -> str:
+    summarized_text = ""
+    filtered_metrics = pre_filter_metrics(metrics=metrics_to_process)
+    if (
+        len(filtered_metrics.metrics_with_only_zero_values)
+        >= len(filtered_metrics.other_metrics) * 0.1
```

Comment:
what if the len condition is false? It seems like you drop the time series entirely from the output!